patches/minecraft/net/minecraft/client/renderer/entity/LivingEntityRenderer.java.patch

```diff
@@ -29,6 +43,18 @@
           f8 = Mth.m_14179_(p_115310_, p_115308_.f_20923_, p_115308_.f_20924_);
           f5 = p_115308_.f_20925_ - p_115308_.f_20924_ * (1.0F - p_115310_);
           if (p_115308_.m_6162_()) {
+@@ -110,8 +_,11 @@
+          }
+       }
+ 
++      this.animator.restoreDefaultPose();
```

Comment:
I don't see the use of setting the default pose still. In this design, it wouldn't matter whether the pose was reset or not. This is because `EntityModel#prepareMobModel` and `EntityModel#setupAnim` are both called directly after making any defaults restored to be a bit pointless. It's more a thing that is accessible to the animator instead of something that would need to be executed imo.

Reply:
The point of the resetting to default pose is because while rotation isn't destructive (destructive in the sense that it negatively influences subsequent animations), changing the rotation axis, part position, etc... Are destructive and need to be reset before each animation frame. You can see this being done in vanilla in certain models. One that comes to mind is the piglin model.

To elaborate further: imagine an arm with rotation axis 0,0,0. For rotation we apply sin/cos so it doesn't matter the previous value.
However, if we move the pivot to 0,10,0 for a specific animation, the next frame after the animation that moved the pivot finishes will have all of the values calculated based on the 0,10,0 pivot.
Ok, we make it up to the 0,10,0 animation to reset it. That means your animation either needs a "resetting" pose at the end or be applied before any and all other animations just for the resetting part.

Reply:
You are describing the model holding state for animations, which is not good since the models are considered singletons in context to their associated renderers. Setting the default pose would only matter if you decided to give the model the ability to hold state which should not be done since you already have an object given to you to retrieve state data from. Additionally, it's called before prepare and setup which would set the rotation data to something other than the default making it a unnecessary call since the data is immediately overwritten before rendering. Resetting the pose of a model is not useful unless the modder desires it that way as part of the animation. Animation is a calculation of knowing the previous location, the new location, and the time delta in between where the locations are stored on the object source itself. So, needing to reset the data when you are just setting information anyways will already happen.

Reply:
Here is a simple example of why this is needed. The animation starts, it sets the pivot point of the player's arm to (0, 10, 0), the animation ends and never runs again. If the pivot point is not overridden by Minecraft's animation setup (usually in `setupAnim`), the arm is now stuck at (0, 10, 0) indefinitely. Minecraft modifies the values on model parts that it needs for the animation, it's not an explicit rotation and position. It may update the `x` and `y` of a pivot point but ignore the `z`; If an animation modifies the `z` value, it will always hold the last value from the animation. It will also be shared to other entities that use the same renderer/model instance.